AdobeStockClient/Model/Client.php

```diff
@@ -126,28 +126,32 @@ public function search(SearchCriteriaInterface $searchCriteria): SearchResultInt
             $searchRequest->setSearchParams($searchParams);
             $searchRequest->setResultColumns($resultColumnArray);
             $client = $this->getConnection()->searchFilesInitialize($searchRequest, $this->getAccessToken());
-            $response = $client->getNextResponse();
+            $searchResult = $this->searchResultFactory->create();
+            $searchResult->setSearchCriteria($searchCriteria);
 
-            $items = [];
-            /** @var StockFile $file */
-            foreach ($response->getFiles() as $file) {
-                $itemData = (array)$file;
-                $itemData['thumbnail_url'] = $itemData['thumbnail_240_url'];
-                $itemData['preview_url'] = $itemData['thumbnail_500_url'];
-                $itemId = $itemData['id'];
-                $attributes = $this->createAttributes('id', $itemData);
+            try {
```

Comment:
It would be better to avoid multiple try-catch levels and separate them to a separate try-catch blocks if possible

Reply:
@sivaschenko Fixed, please check it again